xray/segment.go

```diff
@@ -126,9 +139,28 @@ func BeginSegmentWithSampling(ctx context.Context, name string, r *http.Request,
 		seg.Dummy = true
 	}
 
+	// generates segment and trace id based on sampling decision and secure random env variable
+	idGeneration(seg)
+
 	return context.WithValue(ctx, ContextKey, seg), seg
 }
 
+func idGeneration(seg *Segment) {
+	secureRandom := os.Getenv("AWS_XRAY_SECURE_RANDOM_ID")
+	if secureRandom != "" && strings.ToLower(secureRandom) == "true" {
```

Comment:
Can simplify this condition with adding an `|| seg.Sampled` here right?

Reply:
Yes we can but not sure if that's a good idea. Ideally no-op id generation or not would be based on env variable value. If we include `|| seg.Sampled` here then I think we are using `seg.Sampled` to decide this behavior. Ideally if you look at the code it's a layered condition and I think this would be more safer.